x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -135,26 +135,38 @@ export class MessageSigningService implements MessageSigningServiceInterface {
     return publicKey;
   }
 
-  public async rotateKeyPair(): Promise<boolean> {
-    const isRemoved = await this.removeKeyPair();
-    if (isRemoved) {
+  public async rotateKeyPair(): ReturnType<MessageSigningServiceInterface['rotateKeyPair']> {
+    try {
+      const removeKeyPairResponse = await this.removeKeyPair();
+      if (!removeKeyPairResponse) {
+        return;
+      }
```

Comment:
@dasansol92 @joeypoon So while testing, I noticed that since the return type of `rotateKeyPair` is a `Promise<void>` there's no way to really discern if the function really generated a key pair or skipped the step as there was no previous key pair to delete. 

I strongly feel that we should respond on the API with something like `Previous key pair was not found. Nothing to rotate!`. In that case, we should somehow distinguish this with the return type from `rotateKeyPair`. Probably something like `Promise<void | 'success'>`. Thoughts?

Reply:
I would say, if the key pair was not found, we should throw an error since it should always exists. Other approach is to regenerate it always, even if it didn't exist, so we keep the system consistent. Thoughts?

Reply:
I like the idea of throwing an error. What do you think @joeypoon.

Reply:
5c12c876cb2f6cbaf580ddce728b43f09ec45ad0. Also added error handling on key pair creation 94a8ed996f5018ceba1cb0d1a9336a79a209aec2